### ISSUE TITLE
_processConfig would fail if byteToNormalizedPEM buffer_array is empty array

### ISSUE THREAD

**Issue by Dsstefanov** [hyperledger/fabric-sdk-node#505]
The problem is that buffer array seems to have a different prototype than js definition and when convertBytetoString is called then:

for (const index in buffer_array) {

will start iterating over the prototype properties and result is incorrectly set to be a string of the function definitions instead of being empty string.
Could be fixed by adding a guard:

for (const index in buffer_array) {
			if (buffer_array.hasOwnProperty(index)) {
				const buffer = buffer_array[index];
				const hex_string = buffer.toString(decode_as);
				a_strings.push(hex_string);
			}
		}
		
If the array is not empty then the indexes will be the array indexes (0,1,2...) which will work correctly

**Comment by bestbeforetoday** [comment]
@Dsstefanov Could you confirm whether this is a theoretical issue or something you have actually observed at runtime when the client is processing discovery results?

For reference, the code in question is here:
- https://github.com/hyperledger/fabric-sdk-node/blob/1e5ab02b7dba1470d2c41c345ecbef7f6ad08a56/fabric-common/lib/Utils.js#L556-L575

It does look like the loop over the array could be more safely written using `for...of`, a for loop explicitly using an index variable comparing to `buffer_array.length`, orArray functions like `map()` or `forEach()`.

**Comment by ozer619** [comment]
Hi I'm new to opensource, I would like to work on this issue :)

**Comment by Dsstefanov** [comment]
@bestbeforetoday Yes it is exactly this part of the code. I have observed it from a practical point of view. When I get the response with the config and _processConfig is invoked then because my intermediate certificates are non existent it would fail with (both " _processConfig[mychannel] - Problem with discovery config: Error: Failed to find start line or end line of the certificate." and "No committers assigned to the channel").

**Comment by Dsstefanov** [comment]
@ozer619 

Replacing the code:
for (const index in buffer_array) {
			const buffer = buffer_array[index];
			const hex_string = buffer.toString(decode_as);
			a_strings.push(hex_string);
		}
to:
buffer_array.forEach(buffer => {
			const hex_string = buffer.toString(decode_as);
			a_strings.push(hex_string);
		});
		
will fix the issue.
